generators/client/templates/react/src/main/webapp/app/config/axios-interceptor.ts.ejs

```diff
@@ -27,7 +29,7 @@ const setupAxiosInterceptors = onUnauthenticated => {
       config.headers.Authorization = `Bearer ${token}`;
     }
     config.timeout = TIMEOUT;
-    config.url = `${getBasePath().replace(/\/$/, '')}${config.url}`;
```

Comment:
This regex was changing the URL from `http://localhost:8080/#/` to `http://localhost:8080/#` but it works without the replace, can put back if needed for something

Reply:
I faced some issue coz of this in my own application from which I took these code for JH. I forgot the usecase. I guess it was some edge case. So I guess its better to just be safe here

Reply:
This part of the code seems like a hacky way to do relative links, [`getBasePath()`](https://github.com/jhipster/react-jhipster/blob/f911c0abee3f9f5147051ea09a7e8ee6d287db77/src/util/url-utils.ts#L4) returns the `window.location.href` to get the context path.  Using relative links  (without the leading `/`) achieves the same thing

So with the old code

 - Call to `/api/account`
 - `getBasePath().replace(/\/$/, '')` returns `jhispter.com/sample-app`
 - Request is made to `/api/account` so URL is built as `jhispter.com/sample-app/api/account`

Now:
- Call to `api/account`
- Request is made to `api/account` so URL is built as `jhispter.com/sample-app/api/account`

We use relative links everywhere [in Angular](https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/app/layouts/profiles/profile.service.ts.ejs#L29) and the [docs for SERVER_API_URL](https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/react/webpack/webpack.common.js.ejs#L114-L115) state it should end in `/` or it will use relative links.

Reply:
ok then